app/components/learnergroup-subgroup-list.js

```diff
@@ -1,106 +1,94 @@
-import Component from '@ember/component';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
 import { inject as service } from '@ember/service';
 import { all } from 'rsvp';
-import { task } from 'ember-concurrency';
+import { dropTask } from 'ember-concurrency';
 import pad from 'ember-pad/utils/pad';
 import countDigits from '../utils/count-digits';
 import cloneLearnerGroup from '../utils/clone-learner-group';
 
-export default Component.extend({
-  flashMessages: service(),
-  intl: service(),
-  store: service(),
-  tagName: '',
-  canCreate: false,
-  canDelete: false,
-  isSaving: false,
-  parentGroup: null,
-  saved: false,
-  savedGroup: null,
-  showNewLearnerGroupForm: false,
+export default class LearnergroupSubgroupListComponent extends Component {
+  @service flashMessages;
+  @service intl;
+  @service store;
+  @tracked isSaving = false;
+  @tracked savedGroup;
+  @tracked showNewLearnerGroupForm = false;
+  @tracked currentGroupsSaved = 0;
+  @tracked totalGroupsToSave = 0;
 
-  didReceiveAttrs() {
-    this._super(...arguments);
-    this.setProperties({ saved: false, savedGroup: null });
-  },
+  @action
+  async saveNewLearnerGroup(title) {
```

Comment:
This should be converted into a task otherwise the `this.savedGroup = await newLearnerGroup.save();` asynchronous assignment may fail if the component is torn down before it returns.

Reply:
needs a bigger rewrite then. converting this to a task would result in a task invoking another task which has potential for disaster.

Reply:
✅

Reply:
:heavy_check_mark: